src/commands/create/nft.ts

```diff
@@ -117,54 +113,130 @@ export default class NFT extends Command {
+          message: `\nWould you like to mint the following NFT?\n\n${JSON.stringify(
+            {
+              network: networks[network].shortKey,
+              collectionAddress,
+              tokenId,
+              uriType: UriTypeIndex[uriType],
+              uri,
+            },
+            undefined,
+            2,
+          )}\n`,
+          type: 'confirm',
+          default: false,
+        },
+      ])
+      const mint: boolean = mintPrompt.shouldContinue
+      if (mint) {
+        CliUx.ux.action.start('Minting NFT')
+        receipt = await this.networkMonitor.executeTransaction({
+          network,
+          contract: collection,
+          methodName: 'cxipMint',
+          args: [tokenId, uriType, uri],
+          waitForReceipt: true,
+        })
+        CliUx.ux.action.stop()
       } else {
-        const logs: any[] | undefined = this.networkMonitor.decodeErc721TransferEvent(receipt, collectionAddress)
-        if (logs === undefined) {
-          throw new Error('failed to extract transfer event from transaction receipt')
-        } else {
-          this.log(`NFT has been minted with token id #${logs[2].toString()}`)
-        }
+        this.log('NFT minting was canceled')
+        this.exit()
+      }
+    } else if (collectionType === 'HolographERC721Drop') {
+      const numToMint = await checkNumberFlag(undefined, 'How many NFTs would you like to mint/purchase?')
+      // Connect wallet for signing txns
+      const account = userWallet.connect(this.networkMonitor.providers[network])
+      userWallet.connect(this.networkMonitor.providers[network])
+
+      // Interact with drop contract
+      const drop = new ethers.Contract(collectionAddress, collectionABI, account)
+      const salesConfig = await drop.saleDetails()
+
+      // Confirm if user wants to mint
+      const mintPrompt: any = await inquirer.prompt([
+        {
+          name: 'shouldContinue',
+          message: `\nMinting ${numToMint} NFTs from the following collection: ${await drop.name()} at ${
+            drop.address
+          } for ${ethers.utils.formatEther(salesConfig.publicSalePrice.mul(numToMint).toString())} ${
+            networks[network].tokenSymbol
+          } on ${network}.\n`,
+          type: 'confirm',
+          default: false,
+        },
+      ])
+      const mint: boolean = mintPrompt.shouldContinue
+
+      if (mint) {
+        receipt = await this.networkMonitor.executeTransaction({
+          network,
+          contract: collection,
+          value: salesConfig.publicSalePrice.mul(numToMint), // must send the price of the drop times the number to purchase
+          methodName: 'purchase',
+          args: [numToMint],
+          waitForReceipt: true,
+        })
+        CliUx.ux.action.stop()
+      } else {
+        this.log('NFT minting was canceled')
+        this.exit()
       }
     } else {
-      this.log('NFT minting was canceled')
+      throw new Error('Invalid collection type')
```

Comment:
Should we exit more gracefully here instead of an error?

Reply:
yeah we can do that